janus-idp/backstage-plugins
# feat: add method for `janus-cli package metadata` ; generate updates to package.json for all plugins listed in plugins/plugin-metadata.json

### What does this PR do?

feat: add method for `janus-cli package metadata` [RHIDP-1502](https://issues.redhat.com//browse/RHIDP-1502) ;
generate updates to package.json for all plugins listed in plugins/plugin-metadata.json

Signed-off-by: Nick Boldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit message above for issue number)

### How to test this PR?

* check out PR
* run stuff like thiis:

```
./packages/cli/bin/janus-cli package metadata --help
./packages/cli/bin/janus-cli package metadata --dir plugins/acr --keywords "lifecycle:foo,support:baz,feature:stuff"
cd plugins/argocd; ../../packages/cli/bin/janus-cli package metadata --author foo --license bar --homepage http://bazbar --bugs https://woot
```

Or use the script wrapper to run against a list of plugins:
```
cat ./plugins/plugin-metadata.json
./plugins/plugin-metadata.sh
```

Followup tasks would be to contribute this script / app to upstream community-plugins repo, so we can run it against existing/future plugins and incorporate it into lifecycle automation.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
